src/python/pants/backend/codegen/protobuf/python/python_protobuf_module_mapper.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from typing import Dict, Set
+
+from pants.backend.codegen.protobuf.python.python_protobuf_subsystem import PythonProtobufSubsystem
+from pants.backend.codegen.protobuf.target_types import ProtobufGrcpToggle, ProtobufSources
+from pants.backend.python.dependency_inference.module_mapper import (
+    PythonFirstPartyModuleMappingPlugin,
+    PythonFirstPartyModuleMappingPluginRequest,
+)
+from pants.base.specs import AddressSpecs, DescendantAddresses
+from pants.core.util_rules.stripped_source_files import StrippedSourceFileNames
+from pants.engine.addresses import Address
+from pants.engine.rules import Get, MultiGet, collect_rules, rule
+from pants.engine.target import SourcesPathsRequest, Target, Targets
+from pants.engine.unions import UnionRule
+from pants.util.frozendict import FrozenDict
+from pants.util.logging import LogLevel
+
+
+def proto_path_to_py_module(stripped_path: str, *, suffix: str) -> str:
+    return stripped_path.replace(".proto", suffix).replace("/", ".")
+
+
+# This is only used to register our implementation with the plugin hook via unions.
+class PythonProtobufMappingRequest(PythonFirstPartyModuleMappingPluginRequest):
+    pass
+
+
+@rule(desc="Creating map of Protobuf targets to generated Python modules", level=LogLevel.DEBUG)
+async def map_protobuf_to_python_modules(
```

Comment:
This rule is taken almost verbatim from the 3rdparty mapping rule, outside of changing how the module is generated.